target_decisioning_engine/tests/test_init.py

```diff
@@ -0,0 +1,139 @@
+MOCK_ARTIFACT_RESPONSE_UNSUPPORTED_VERSION = HTTPResponse(status=200, body=json.dumps(ARTIFACT_UNSUPPORTED_VERSION))
+
+
+class TestTargetDecisioningEngine(unittest.TestCase):
+
+    def setUp(self):
+        self.decisioning = None
+
+    def tearDown(self):
+        if self.decisioning:
+            self.decisioning.stop_polling()
+
+    def test_initialize(self):
+        config = deepcopy(CONFIG)
+        config.polling_interval = 0
+        self.decisioning = TargetDecisioningEngine(config)
+
+        with patch.object(PoolManager, "request", return_value=MOCK_ARTIFACT_RESPONSE):
+            self.decisioning.initialize()
+
+        self.assertEqual(self.decisioning.artifact, ARTIFACT_BLANK)
+        self.assertIsNotNone(self.decisioning.artifact_provider)
+        self.assertEqual(self.decisioning.artifact_provider.subscription_count, 1)
+
+    def test_initialize_updates_artifact_on_polling_interval(self):
+        config = deepcopy(CONFIG)
+        config.polling_interval = 1  # seconds
+        self.decisioning = TargetDecisioningEngine(config)
+
+        with patch("target_decisioning_engine.artifact_provider.get_min_polling_interval", return_value=0):
+            with patch.object(PoolManager, "request", return_value=MOCK_ARTIFACT_RESPONSE) as artifact_request_mock:
+                self.decisioning.initialize()
+
+                time.sleep(5)
+
+                self.assertGreaterEqual(artifact_request_mock.call_count, 4)
+                artifact = self.decisioning.get_raw_artifact()
+                self.assertIsNotNone(artifact)
+
+
+    def test_initialize_error_fetching_artifact(self):
+        config = deepcopy(CONFIG)
+        config.polling_interval = 0
+        config.event_emitter = Mock()
+        self.decisioning = TargetDecisioningEngine(config)
+
+        with patch.object(PoolManager, "request", return_value=MOCK_ARTIFACT_RESPONSE_BAD):
+            with patch("target_decisioning_engine.artifact_provider.BACKOFF_FACTOR", 0):
+                with self.assertRaises(Exception) as err:
+                    self.decisioning.initialize()
+                    self.assertEqual(json.loads(err.exception.body),
+                                    {"status": 500, "message": "The decisioning artifact is not available"})
+                    self.assertEqual(config.event_emitter.call_count, 11)  # validate retries
+                    self.assertEqual(config.event_emitter.call_args[0][0], ARTIFACT_DOWNLOAD_FAILED)
+
+    def test_get_offers(self):
+        config = deepcopy(CONFIG)
+        config.polling_interval = 0
+        self.decisioning = TargetDecisioningEngine(config)
+
+        with patch.object(PoolManager, "request", return_value=MOCK_ARTIFACT_RESPONSE):
+            self.decisioning.initialize()
+
+        get_offers_opts = TargetDeliveryRequest(request=TARGET_REQUEST, session_id="dummy_session")
+        offers = self.decisioning.get_offers(get_offers_opts)
+        self.assertIsNotNone(offers)
```

Comment:
offers is just a mock, so the intent isn't to validate the data itself.  The purpose of the test is just to make sure when calling self.decisioning.initialize() it sets the internal self.artifact class member so that when get_offers() is called the artifact is available.